### ISSUE TITLE
Some IP in sub-net are not accessible inside wsl2

### ISSUE THREAD

**Issue by dhruvinsh** [microsoft/WSL#5427]
I can ping IP address from command prompt, but the same IPs are not accessible within WSL2.

Command Prompt vs WSL2
![image](https://user-images.githubusercontent.com/33428164/84900041-88be6f80-b077-11ea-9b8f-ff3366cf96b0.png)

EDIT:
Corrected screenshot:
![image](https://user-images.githubusercontent.com/33428164/84933779-68f17080-b0a4-11ea-9636-e3c64d91ffd3.png)

**Comment by MarceloMPJ** [comment]
New update resolve this.

**Comment by dhruvinsh** [comment]
Hi @MarceloMPJ new update of what?

I checked Microsoft Store there is none. can you please elaborate regarding update? and how can i get it?

**Comment by therealkenc** [comment]
Can't follow the screencap; the windows address is `10.254.90.25` and the unreachable one in WSL is `10.254.90.215`. Those are not the same IPs. Maybe do one like using interop this with your fail address.

![image](https://user-images.githubusercontent.com/7049700/84921954-01263000-b07a-11ea-9f60-c7d260167a97.png)

**Comment by dhruvinsh** [comment]
My bad,
Thank you pointing it out, Here is corrected screenshot,
![image](https://user-images.githubusercontent.com/33428164/84933737-57a86400-b0a4-11ea-8701-52c51cad11c5.png)

**Comment by therealkenc** [comment]
Alright. Needless to say this isn't going to be able to be reproduced locally, so the question is how to track down what is happening for you on that address. We are almost certainly going to arrive at "for by-design reason" or externality (if we arrive anywhere) but I can't off the cuff guess what that might be.

As an initial data point, post the output of `/bin/ping -V` so it is on the books. There is more than one kind of `ping` and the behavior of a given implementation isn't the same. Also do a `cat /proc/version` (per the previous screencap), `cmd.exe /c ver`, and `lsb_release -r`.

Next, identify what is different about your `.25` destination and your `.215` destination. [They are different _somehow_ by definition.] 

Disable Windows firewall temporarily just to get that variable out of the way.

Next thing to try is go root with `sudo su` and run `ping` from the root prompt.

If that doesn't work, about the only thing left is to fire up [Wireshark](https://www.wireshark.org/) on Windows and look at the traffic, and see what is going out your ethernet interface. We know _something_ is going out, or your `pings` wouldn't be working at all under WSL (and this would be a different exercise). That might shed some light on why nothing is coming back.

**Comment by dhruvinsh** [comment]
```
> /bin/ping -V
ping from iputils s20190709
```

```
> cat /proc/version
Linux version 4.19.104-microsoft-standard (oe-user@oe-host) (gcc version 8.2.0 (GCC)) #1 SMP Wed Feb 19 06:37:35 UTC 2020
```

```
> cmd.exe /c ver
Microsoft Windows [Version 10.0.19041.264]
```

```
> lsb_release -r
Release:        20.04
```

> Disable Windows firewall temporarily just to get that variable out of the way.
I tried, but still the same, i can not ping .215 from WSL

> Next thing to try is go root with sudo su and run ping from the root prompt.
tried, still the same

> wireshark:
As you can see, very first 2 packets were for `.215` and other two were for `.25` where i received a reply back.
![image](https://user-images.githubusercontent.com/33428164/84943194-02734f00-b0b2-11ea-9262-0ed7b9af3af3.png)

**Comment by dhruvinsh** [comment]
@therealkenc thanks for the efforts, Issue is resolved,

The edge device gateway did not get apply properly. Setting up proper IP route fixed the issue.

**Comment by walwari** [comment]
> @therealkenc thanks for the efforts, Issue is resolved,
> 
> The edge device gateway did not get apply properly. Setting up proper IP route fixed the issue.

I have same issue, Is it possible to share more details above resolve?

**Comment by dhruvinsh** [comment]
so my windows IP was 10.254.90.x and gateway was 10.254.90.1.

WSL IP was 172.x.x.x.

Remote devices IP were 10.254.90.25 and 10.254.90.215. but here is the thing, first device had a gateway set to 10.254.90.1 and second device did not have any gateway.

Since windows PC was on the same subnet as above device, I was able to ping it, Gateway wasn't required.

But think of WSL as some another device which is bridged via windows network, and I need to have proper gateway to the remote device. Once I did it the issue got resolved.
